lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -7579,6 +7579,11 @@ bool SwiftASTContext::DumpTypeValue(
 
   swift::CanType swift_can_type(GetCanonicalSwiftType(type));
 
+  if (auto *dyn_self = llvm::dyn_cast_or_null<swift::DynamicSelfType>(
+          swift_can_type.getPointer()))
```

Comment:
Should this just be part of GetConanicalSwiftType?

Reply:
Canonical types have a pretty specific meaning I think, it's just a type without any sugar. We could change it in SwiftASTContext, but then we won't really have what is referred as a canonical type in the compiler.